### PR TITLE
feat(AP-3822): Add Rollup with TypeScript and Sass

### PR DESCRIPTION
**Pull request by joshua-barnard-evolv**
Refactored the codebase to use Rollup with TypeScript and Sass.

### CODE REVIEW

**Comment by joshua-barnard-evolv** [comment]
I pushed this on the wrong branch, but I can't for the life of me fix it.

File: package.json [hunk 1]
```diff
@@ -0,0 +1,39 @@
+{
+  "name": "chrome-extension",
+  "version": "1.0.0",
+  "description": "## Getting Started --- #### **1)** In this Github project, click the green `Code` button and choose `Download ZIP`.  Save the zip file somewhere that you're not going to delete it by accident.  Once downloaded, open the zip file and you should see a directory named `evotools-main`. <img src=\"https://imgur.com/8hvUBj4.png\"/>",
+  "main": "src/background.js",
+  "type": "module",
+  "scripts": {
```
Comment (mattstrom) [inline package.json:7]: To prevent accidental publishing
```suggestion
  "private": true,
  "scripts": {
```

File: package.json [hunk 2]
```diff
@@ -0,0 +1,39 @@
+{
+  "name": "chrome-extension",
```
Comment (mattstrom) [inline package.json:2]: Not necessary here, but I like to add a scope to our packages
```suggestion
  "name": "@evolv/chrome-extension",
```

File: src/popup.ts [hunk 3]
```diff
@@ -284,13 +276,13 @@ const handleClearSelectionClicks = async () => {
...
 
-const setBlockExecutionStatus = (blockExecutionValue) => {
+const setBlockExecutionStatus = (blockExecutionValue: BlockExecution) => {
   waitForElement("#block-execution-toggle input").then(function (toggleInput) {
     blockExecutionValue
       ? toggleInput.checked = false
       : toggleInput.checked = true;
 
-    toggleInput.addEventListener('click', function (e) {
+    toggleInput.addEventListener('click', function (e: any) {
```
Comment (mattstrom) [inline src/popup.ts:285]: You'll have to add `"dom"` to _tsconfig.json_ as a lib for this to work.
```suggestion
    toggleInput.addEventListener('click', function (e: MouseEvent) {
```

File: src/popup.ts [hunk 4]
```diff
@@ -1,31 +1,23 @@
...
 
-const sendMessage = function (message) {
+const sendMessage = function (message: any) {
   chrome.tabs.query({
     currentWindow: true,
     active: true
   }, function (tabs) {
-    var activeTab = tabs[0];
-    chrome.tabs.sendMessage(activeTab.id, message);
+    var activeTab: chrome.tabs.Tab = tabs[0];
```
Comment (mattstrom) [inline src/popup.ts:17]:
```suggestion
    const activeTab: chrome.tabs.Tab = tabs[0];
```

File: src/injectScript.ts [hunk 5]
```diff
@@ -1,13 +1,17 @@
+import { Evolv, Experiments } from './types';
+
+
+let evolv: Evolv = (window as any).evolv;
```
Comment (mattstrom) [inline src/injectScript.ts:4]: When `dom` has been added:
```suggestion
let evolv: Evolv = (window as Window).evolv;
```

**Review by mattstrom — APPROVED** [review]

**Pull request merged**